thrift.go

```diff
@@ -0,0 +1,169 @@
+package baseplate
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/getsentry/raven-go"
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/reddit/baseplate.go/batcherror"
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+type baseplateThriftServer struct {
```

Comment:
I don't think we need this, I think what we want is more like:

```
func main() {
    var closers baseplate.Closers
    defer closers.Cleanup()

    baseplate := SetupBaseplateStruct()
    closers = baseplate.start()
    server, err := thriftbp.NewBaseplateServer(baseplate, other_args...)
    if err {panic}
    return log.Fatal(server.serve())
}
```

where `main` is defined in each service.